### PR TITLE
Able to download the attachment from form in form filling state + new styling

### PR DESCRIPTION
**Pull request by lassopicasso**
## Description
User can now download attachment during form filling state - the attachment-link are not clickable until the upload is finished.

The attachment lists (+ dropzoneComponent) got some new design.

Done some css cleanup and refactoring (created a new file "AttachmentFileName" for the attachment-link + doc icon) .

In the future I would like to see that the two components FileUpload (without tag) and FileList (with tag) can reuse the same parts of table, and with that also reuse more of the styling. Atm there are a lot of duplicated styling for these two components.

Did some pair programming with @Magnusrm where we added a cypress test that checks if the uploaded file can also be downloaded. Had to create a workaround to make sure the page reloads when you click the attachment link (otherwise Cypress would wait for a non-existent page to reload when clicking the anchor)

```  
    const loadScript = '<script> setTimeout(() => location.reload(), 1000); </script>';
    cy.get('body').invoke('append', loadScript);`
```

> WCAG - I am not 100% sure if I've used the arias correctly for such elements as icons etc, appreciate any feedback about these.

## Related Issue(s)

- closes #https://github.com/Altinn/app-frontend-react/issues/140

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [ ] Unit test(s) have been added/updated
  - [x] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [x] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [ ] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [x] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Changes/additions to component properties
  - [ ] Changes are reflected in both `src/layout/layout.d.ts` and `layout.schema.v1.json`, and these are all backwards-compatible
  - [x] No changes made
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [x] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

File: src/layout/FileUploadWithTag/FileListComponent.module.css [hunk 1]
```diff
@@ -67,23 +64,23 @@
...
   font-family: 'Altinn-DIN', sans-serif;
   font-weight: 400;
+  font-size: 1rem;
   border-bottom: 1px dotted var(--border-color-primary);
   color: #000;
 }
+.tableBody tr td:last-child > button {
+  float: right;
+  padding-right: 0;
+}
```
Comment (bjosttveit) [inline src/layout/FileUploadWithTag/FileListComponent.module.css:80]: same as above

**Review by bjosttveit — APPROVED** [review]
Looking good! I looked into the cypress download thing out of curiosity, and I guess that is what people do to get around it 🤷

**Comment by Magnusrm** [comment]
Hoping you will forgive me for merging this @lassopicasso, while you're recovering 🙏 😁

**Comment by lassopicasso** [comment]
> Hoping you will forgive me for merging this @lassopicasso, while you're recovering 🙏 😁

Haha, np. Thx for the help :)

**Pull request merged**